### PR TITLE
[SDESK-1184][SDESK-1033] Adding content api search page.

### PR DESCRIPTION
**Pull request by mdhaman**
Include changes to `sdSearchPanel` , `sdSearchParameters` and sort so that it can be reused in content api search page.

Depends https://github.com/superdesk/superdesk-core/pull/940

### CODE REVIEW

**Comment by mdhaman** [comment]
@petrjasek @akintolga review pls

File: scripts/apps/content-api/directives/ContentAPISearchPanelDirective.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+     * @name sdContentApiSearchPanel#search
+     * @description broadcast 'search:parameters' event to trigger search.
+     */
+    search() {
+        this.scope.$broadcast('search:parameters');
+    }
+
+    /**
+     * @ngdoc method
+     * @name sdContentApiSearchPanel#search
```
Comment (akintolga) [inline scripts/apps/content-api/directives/ContentAPISearchPanelDirective.js]: method name is wrong here in the docs

File: scripts/apps/publish/services/SubscribersService.js [hunk 2]
```diff
@@ -46,6 +49,16 @@ export function SubscribersService(api, $q, $filter) {
                 parts.push({_id: id});
             });
             return this.fetchSubscribers({$or: parts});
+        },
+        initialize: function() {
+            if (!this.fetched) {
+                this.fetched = _getAllSubscribers()
+                    .then((subscribers) => {
+                        this.subscribers = subscribers;
+                        this.subscribersLookup = _.keyBy(subscribers, '_id');
```
Comment (akintolga) [inline scripts/apps/publish/services/SubscribersService.js]: 👍 for `_.keyBy`

File: scripts/apps/search/tests/search.spec.js [hunk 3]
```diff
@@ -116,15 +116,20 @@ describe('search service', () => {
...
 
-    it('can sort items', inject((search, $location, $rootScope) => {
-        search.setSort('urgency');
+    it('can sort items', inject((search, sort, session, $location, $rootScope) => {
+        sort.setSort('urgency', search.sortOptions);
         $rootScope.$digest();
         expect($location.search().sort).toBe('urgency:desc');
-        expect(search.getSort()).toEqual({label: 'Urgency', field: 'urgency', dir: 'desc'});
+        session.identity = {_id: 'foo'};
+        var criteria = search.query($location.search()).getCriteria();
```
Comment (akintolga) [inline scripts/apps/search/tests/search.spec.js]: `let` could be used here

File: scripts/apps/search/services/SearchService.js [hunk 4]
```diff
@@ -401,7 +376,7 @@ export function SearchService($location, gettext, config, session, multi, prefer
          */
         this.getCriteria = function getCriteria(withSource) {
             var search = params;
-            var sort = getSort();
+            var sort = sortService.getSort(sortOptions);
```
Comment (akintolga) [inline scripts/apps/search/services/SearchService.js]: `let` could be used here

**Pull request merged**